### PR TITLE
Todo list add remove

### PR DESCRIPTION
**Pull request by AbuTalha3**
In this PR I have:-
- Adding and Removing functions are correctly working.
- On page load, it renders the dynamically created list of tasks in the dedicated placeholder.
- Removing of completed tasks can be done after selecting the check box.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1 +1,124 @@
...
+      text: 'Car wash',
+      completed: false,
+      index: 2,
+    },
+    {
+      text: 'Do the dishes',
+      completed: false,
+      index: 3,
+    },
+  ];
```
Comment (Johnadibe) [inline src/index.js]: > - [ ] According to the project requirement, kindly remove all hardcoded items from the tasks array. This adjustment will enhance the user experience and ensure smooth navigation within the application. 👍
> 
> remove
>

File: modules/main.js [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+      index: todos.length + 1,
+    };
+
+    todos.push(todoItem);
+    saveTodosToLocalStorage();
+
+    textInputField.value = '';
+
+    renderTodoItem(todoItem);
+  });
```
Comment (Johnadibe) [inline modules/main.js]: > - [ ] According to the project requirement from the image below, I've noticed that when I include a new item in the to-do list, it doesn't get updated in Local Storage. Could you please address this issue? 👍 It's important to resolve this problem as having the to-do list synchronized with Local Storage ensures that the user's tasks are saved even if they close or refresh the app. This offers a seamless and consistent experience, where tasks remain accessible across different sessions.😊
> 
> - [ ] Also, When adding new tasks, they should automatically have the "completed" property set to "false" and the "index" property set to match the new array's length. Additionally, when a task is deleted, the indexes of the remaining tasks should be updated to reflect the current order and maintain uniqueness.👍
> 
> top
> 
> [screen-capture.webm](https://github.com/AbuTalha3/TodoList-RM/assets/40009354/e276aaf7-a98d-4a75-9915-ea6eeef652bf)
>

**Review by Johnadibe — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES ♻️
### Hi @AbuTalha3👋🏽, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

mario_ladders

## Highlights
✅ You used GitHub Flow correctly 👍🏽
✅ Linter checks passed✔️
✅ Users can add a new task to the array✔️
✅ Users can delete a task from the array✔️
✅ Users can edit task descriptions✔️


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Efiamotu-1 — APPROVED** [review]
Hi @AbuTalha3 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# STATUS: APPROVED 🕺🥳 

image

## Highlights :dart: 
* Correct GitHub flow :heavy_check_mark: 
* All linters passing :heavy_check_mark: 
* Good PR summary :heavy_check_mark: 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, you can now Merge since it is Approved ✅, However if there are any conflicts please resolve them before merging.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**